### DISCUSSION TITLE
How to create a repeatable job with startDate. ? Is startDate available.

### DISCUSSION THREAD

**Discussion by mai1x9** [taskforcesh/bullmq#2973] [General]
Hi,

I want to create repeatable job every 1 minute, but run at 3 hours from now in future and subsequently follow every 1 minute. However the job is getting triggered immediately every minute.

- I refered: https://github.com/OptimalBits/bull/issues/1786 , however this seem not working. I have created a nee queue and tried still not working.
- I tried just startDate, still even that is not working. As mentioned here, https://github.com/OptimalBits/bull/issues/1648#issuecomment-1159763859, there is no startDate and @manast suggested to compute subtraction and create delayed job. This contradicts to https://github.com/OptimalBits/bull/issues/1786#issuecomment-792257144
> Delay cannot be combined currently with repeatable jobs though.

So the whole point is, the `startDate` does not work, and it was suggested to use `delay` and when `delay` used, we cannot use with cron patterns and to run cron patterns at future use `startDate` -> **A contradiction**

**Comment by manast** [comment]
Which version of BullMQ are you using? I am confused because you are referring to a different repo than this one...

**Reply by manast** [reply]
Btw, there is no contradiction, you are mixing two different features, repeatable jobs and delayed jobs. What I say about delayed jobs is true and works, for repeatable jobs startDate works too, there is no contraction. But in any case, you are referring to issues from a different repo.

**Reply by mai1x9** [reply]
Thanks for quick reply @manast 
I am using bullmq (not bull, yes there are different repos, I thought similar options would be available).

I am using bullmq version: `5.12.12`.

> What I say about delayed jobs is true and works, for repeatable jobs startDate works too, there is no contraction.

However in my code this is not working, as mentioned when `startDate` and cron pattern used. Cron is triggered immediately instead of after `startDate`

**Reply by manast** [reply]
Please test with the latest version and if anything is not working provide a test case in a new issue and we will certainly look into it.
